test/es6/dynamic-module-functionality.js

```diff
@@ -0,0 +1,359 @@
+                    (v)=>{
+                        assert.areEqual('function', typeof v.default, 'typeof default export from ModuleDefaultExport2.js is function');
+                    },
+                    (err)=>{ assert.isTrue(false, err.message); }, _asyncEnter, _asyncExit);
+                `;
+
+            testScript(functionBody, "Typeof a module export", false, true);
+            testModuleScript(functionBody, "Typeof a module export", false, true);
+        }
+    },
+    {
+        name: "Circular module dependency",
+        body: function () {
+            let functionBody =
+                `testDynamicImport(
+                    ()=>import('ModuleCircularFoo.js'),
+                    (v)=>{
+                        v.reset();
+                        assert.areEqual(2, v.circular_foo(), 'This function calls between both modules in the circular dependency incrementing a counter in each');
+                        assert.areEqual(4, v.rexportbar(), 'Second call originates in the other module but still increments the counter twice');
+                    },
+                    (err)=>{ assert.isTrue(false, err.message); }, _asyncEnter, _asyncExit);
+                `;
+
+            testScript(functionBody, "Circular module dependency", false, true);
+            testModuleScript(functionBody, "Circular module dependency", false, true);
+
+        }
+    },
+    {
+        name: "Implicitly re-exporting an import binding (import { foo } from ''; export { foo };)",
+        body: function () {
+            let functionBody =
+                `testDynamicImport(
+                    ()=>import('ModuleComplexReexports.js'),
+                    (v)=>{
+                        assert.areEqual('foo', v.foo(), 'Simple implicit re-export');
+                        assert.areEqual('foo', v.baz(), 'Renamed export imported and renamed during implicit re-export');
+                        assert.areEqual('foo', v.localfoo(), 'Export renamed as import and implicitly re-exported');
+                        assert.areEqual('foo', v.bar(), 'Renamed export renamed as import and renamed again during implicit re-exported');
+                        assert.areEqual('foo', v.localfoo2(), 'Renamed export renamed as import and implicitly re-exported');
+                        assert.areEqual('foo', v.bar2(), 'Renamed export renamed as import and renamed again during implicit re-export');
+                    },
+                    (err)=>{ assert.isTrue(false, err.message); }, _asyncEnter, _asyncExit);
+                `;
+
+            testScript(functionBody, "Implicitly re-exporting an import binding (import { foo } from ''; export { foo };)", false, true);
+            testModuleScript(functionBody, "Implicitly re-exporting an import binding (import { foo } from ''; export { foo };)", false, true);
+        }
+    },
+];
+
+testRunner.runTests(tests, { verbose: WScript.Arguments[0] != "summary" });
```

Comment:
Nest import syntax is test in dynamic-module-import-specifier.js


---
In reply to: [115367185](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115367185) [](ancestors = 115367185)